pkg/operator/mon/mon.go

```diff
@@ -378,24 +378,8 @@ func (c *Cluster) assignMons(mons []*monConfig) error {
 		var node v1.Node
 		// when there are enough available nodes for the mon size we just pick the next empty one
 		if len(availableNodes) >= c.Size {
-			for {
-				// pick one of the available nodes where the mon will be assigned
-				node = availableNodes[nodeIndex]
-				if c.HostNetwork {
-					// safety precaution in case the nodeIndex should ever be higher
-					//than the availableNodes
-					if nodeIndex >= len(availableNodes) {
-						return fmt.Errorf("no more nodes available for hostNetwork and mon %s", m.Name)
-					}
-					if _, ok := c.mapping.Port[node.Name]; ok {
-						nodeIndex++
-					} else {
-						break
-					}
-				} else {
-					break
-				}
-			}
+			// pick one of the available nodes where the mon will be assigned
+			node = availableNodes[nodeIndex]
```

Comment:
could you remove the if/else statement completely? This line already includes both if/else scenarios, right?
```
node = availableNodes[nodeIndex%len(availableNodes)]
```

Reply:
@travisn Oh sorry, I completely overthought the logic here while I made my changes. I removed most code there only the code for the port increase (which is required for `hostNetwork` mode) when the node is already used, remains now.

Please let me know if it's good now, then I'll rebase and squash :)